### DISCUSSION TITLE
Databases panel width & icon clarity

### DISCUSSION THREAD

**Discussion by Luthien-in-edhil** [pawelsalawa/sqlitestudio#3982] [General]
Thanks for this very useful tool! I love it.

For what it's worth, here's some minor issues I found using it:

- width of the Databases area on the left: it's so narrow that it obscures the names of the tables and views. I can of course resize it, but whenever I click in that area it 'snaps back' to that initial size;
- the icons are really small compared to the rest of the UI and it's not very clear what they refer to

<img width="783" alt="SQLiteStudio" src="https://user-images.githubusercontent.com/446688/108595199-e6c9f200-737e-11eb-98b3-b0cfbcc3b4a2.png">

Cheers
Lúthien

**Comment by pawelsalawa** [comment]
First problem is already addressed for upcoming 3.3.0 with fix #3421.

Second problem - icons are small due to 2 reasons:
1. Users often requested to maximize the "workspace" area, as data view is critical part of any app working with databases.
2. The icon set that I use has mostly 16x16 icons (raster icons, not vector). I haven't found any good, free and as rich (in number of icons) iconset that would have bigger icons or be vector-based (scalable). I've made an attempt once with some publicly available SVG icons already, but these days all SVG icon sets are "flat" and made mostly for web, not for desktop apps.

Having that said, I have few good news:
- you can always find out what the icon refers to by holding up mouse over it - they all have tooltips.
- number if buttons on the main toolbar will be significantly reduced in 3.3.0 thanks to #3975
- also in version 3.3.0 a HighDPI support is improved (it's not applicable in your case, I believe, but still...), so on high dpi devices these buttons will be finally usable

**Reply by Luthien-in-edhil** [reply]
Super, thanks!

I think that my laptop is high DPI though, it's a late 2019 16'' macbook, screen resolution is 3072 × 1920.

If you like and it is any help, I could have a go at creating a set of larger / vector based icons based on the current ones?

**Reply by pawelsalawa** [reply]
In that case indeed you have high-dpi screen, although icons don't seem to be that really small on your screenshot. I'm not sure if 3.3.0's HighDPI enhancement will change anything for you. Let's hope it will.

Regarding icons - that would be awesome! At the same time, be aware, that the whole iconset has around 3500 icons... (it's Fugue Icons https://p.yusukekamiyamane.com/). While currently I use around 200 icons in the application, I always have all 3500 at my disposition, whenever I want to add new icon to the app. That could be a great challenge to you. Just saying.

**Reply by Luthien-in-edhil** [reply]
Yeah - 3500 is bit much 😨. From looking at the app I would have thought it would be about 20 or so!

I suppose [these](https://github.com/pawelsalawa/sqlitestudio/tree/master/SQLiteStudio3/guiSQLiteStudio/img) the ones currently in use?  

I got curious because so much good free stock photo's and illustrations have become available through sites like unsplash, pexels and pixabay, so I wondered if this also went for icons. So if the last time you looked was a couple of years ago, there might now be a lot more, like [flaticon](https://www.flaticon.com/free-icon/frequency_2289375?term=tables&page=1&position=35&page=1&position=35&related_id=2289375&origin=search) or [streamlineicons](https://streamlineicons.com/download/) or [freeicons](https://freeicons.io/filter/popular/all/table) and several others by the look of it ... 
It looks like they're all free but require attribution like yusukekamiyamane.

Example - [add column](https://www.flaticon.com/free-icon/column_3602051?term=table+column&related_id=3602051&origin=search) - I think the most challenging part is to find a matching set. 

I don't think it would be too hard to do that for the ~ 200 ones, especially not if part of those can be found free to use and there's not a great hurry. 

Would it be an idea if I would get a minimal set together so that you can have a test and see if it looks good in the UI?

**Reply by pawelsalawa** [reply]
I'm aware of pixabay, unsplash, ... The problem with using custom icons from portals like this is that you will never collect consistent set of icons for all purposes needed. There will always be something missing and I'm not good at graphics. Not at all.

By having 3500 icons within one theme available, I have planty to choose from.

**Reply by Luthien-in-edhil** [reply]
I see your point, yes.

Anyhow, if the situation arises you need some custom icons free to ask!
